src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -1878,7 +1878,7 @@ public struct ReadOnlyMemory<T>
         public bool DangerousTryGetArray(out ArraySegment<T> arraySegment) { throw null; }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-    public partial struct ReadOnlySpan<T>
+    public readonly ref struct ReadOnlySpan<T>
```

Comment:
@VSadov When I tried to do `partial readonly ref` or `readonly ref partial` here, I got very weird error messages.

Reply:
readonly/public are modifiers and can be in any order, partial ref are contextual and must be immediately before struct.  
Try "public readonly partial ref struct"

Will need to see if error can be clearer.